toolkit/event-observer/src/main/kotlin/br/com/arch/toolkit/util/_dataResult.kt

```diff
@@ -104,24 +103,34 @@ fun <T> dataResultNone() = DataResult<T>(null, null, NONE)
 
 //region Transformation Methods
 /**
- * Merges a DataResult<T> with a DataResult<R>
+ * Transforms a [DataResult] with a pair of nullable values to a [DataResult] with non-null values.
+ *
+ * If any value in the pair is null, the result will be null.
+ *
+ * @return A [DataResult] with a pair of non-null values, or null if any value is null.
+ */
+fun <T, R> DataResult<Pair<T?, R?>>.toNotNull(): DataResult<Pair<T, R>>? = when (val data = data) {
```

Comment:
Esse nome ficou estranho, devia ser um toNotEmpty também porque se o conteúdo for null ele volta null